go/libkb/secret_store_file.go

```diff
@@ -37,44 +65,122 @@ func (s *SecretStoreFile) RetrieveSecret(username NormalizedUsername) (LKSecFull
 	return newLKSecFullSecretFromBytes(secret)
 }
 
+func (s *SecretStoreFile) retrieveSecretV2(username NormalizedUsername) (LKSecFullSecret, error) {
+	xor, err := ioutil.ReadFile(s.userpathV2(username))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return LKSecFullSecret{}, ErrSecretForUserNotFound
+		}
+
+		return LKSecFullSecret{}, err
+	}
+
+	noise, err := ioutil.ReadFile(s.noisepathV2(username))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return LKSecFullSecret{}, ErrSecretForUserNotFound
+		}
+
+		return LKSecFullSecret{}, err
+	}
+
+	secret, err := s.noiseXOR(xor, noise)
+	if err != nil {
+		return LKSecFullSecret{}, err
+	}
+
+	return newLKSecFullSecretFromBytes(secret)
+}
+
+func (s *SecretStoreFile) noiseXOR(secret, noise []byte) ([]byte, error) {
```

Comment:
Can we introduce `type`s for the `[]byte` to make sure we don't have any mistaken argument swaps?

Reply:
done